### PR TITLE
Make note of doc duplication with reexports

### LINKED ISSUE #16645: Document why std docs have many apparent duplicates
[Reddit thread](http://www.reddit.com/r/rust/comments/2e47xy/duplication_in_the_standard_library/):
> Why are data structures generally duplicated in the rust standard library, or at least apparently so in the API documentation?

> e.g. std::string::String vs. collections::string::String, std:fmt::String vs. core::fmt::String, std::vec::Vec vs. collections::vec::Vec?

I'm not certain which of these the user was referring to but both could be possibly weird at first appearance:
- Why is `String` in `std::string` and `collections::string` and others
- Why is `String` in `string` and `Vec` in `vec`

It was suggested that this be included in a **How to read this documentation** somewhere. If it was written, I would imagine it should be automatically included in the rust std documentation system so that it is included in all rust derived projects like **gfx-rs** and **piston** documentation.

cc: @steveklabnik

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #16645

Fixing this in any deeper way will require an RFC, so let's just document the current behavior.

### CODE REVIEW

File: src/doc/trpl/documentation.md [hunk 1]
```diff
@@ -294,3 +294,20 @@ Documentation` on the first line).
...
+
+pub use foo::bar;
+```
+
+This will create documentation for `bar` both inside the documentation for
+the crate `foo`, as well as the documentation for your crate. It will use
+the same documentation in both places.
+
+In the standard library, this leads to much documentation that's shown in
+both `std` and `core`, since `std` re-exports a lot of `core`.
```
Comment (alexcrichton) [inline src/doc/trpl/documentation.md]: While true, this doesn't necessarily seem like it belongs in the docs. I would expect this to be more of a bug report then anything else.

**Comment by alexcrichton** [comment]
This may also want to mention `doc(no_inline)`

**Comment by steveklabnik** [comment]
@alexcrichton updated!

**Pull request merged**